share/zfsnap/core.sh

```diff
@@ -203,6 +203,34 @@ GreaterDate() {
     return 0
 }
 
+#Find common snapshot from two lists of snapshot names
+FindCommonSnapshot(){
+	local local_snaps=$1
+	local remote_snaps=$2
+	
+	#test to see if snapshot names were provided
+	if [[ -n "${local_snaps[*]}" && -n "${remote_snaps[*]}" ]]; then
```

Comment:
Overall, if `ListLocalSnapshots` and `ListRemoteSnapshots` simply acquired the output from `zfs list -d 1 -t snapshot -S creation -H`, then the entire logic could be taking the most recent `remote_snap`, checking if it's in `local_snaps`, and then you know.

    remote